autogen.sh

```diff
@@ -1,9 +1,8 @@
+#! /bin/sh -x
```

Comment:
Surprised shellcheck did not catch how this should be a `/usr/bin/env <shell>` to be portable.

Reply:
I thought posix shell scripts ought to always be `#!/bin/sh` but [apparently that's not so](https://news.ycombinator.com/item?id=17069408). Regardless, there's no point in a shebang in the middle of the file... Or so I thought?

```
$ ./autogen.sh
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
^C
```

Fun exercise to try to explain to yourself why this works at all; the kernel is supposed to read the first two bytes of the file and look for `#!` to determine the interpreter. So why does this work?

```
$ strace ./autogen.sh
execve("./autogen.sh", ["./autogen.sh"], 0x7fff933c9790 /* 151 vars */) = -1 ENOEXEC (Exec format error)
strace: exec: Exec format error
+++ exited with 1 +++
```

Answer found here: http://www.faqs.org/faqs/unix-faq/faq/part3/section-16.html